### ISSUE TITLE
Hourly Bells - Not Working on karazhan.

### ISSUE THREAD

**Issue by LordWaycrest** [TrinityCore/TrinityCore#23638]
**Description:**

The hourly bells event does not work at all.

**Current behaviour:**

When the clocks strike the start of another hour the clock bells in Stormwind, Orgrimmar, Darkshire, etc.

**Expected behaviour:**

The hourly bells event should start at the start of the new hour. (E.g 12:00 AM)

**Steps to reproduce the problem:**

1. Go to a town or city like Stormwind or Darkshire
2. Wait till the clock strikes a new hour
3. Find that the hourly bells event does not work

**Branch(es):**

3.3.5a

**TC rev. hash/commit:** 

TrinityCore rev. c27804473976+ 2019-07-15 22:28:10 +0200 (3.3.5 branch) (Win64, RelWithDebInfo, Static)

**Operating system:**

Windows Server 2016 Standard

NOTE: The event does start at the right time and end at the right time however, no bells ring

**Comment by Kittnz** [comment]
whats up with these events breaking? :/

**Comment by illfated** [comment]
Maybe the sound files are not linked correctly or the linking has changed?

**Comment by LewisPotato** [comment]
> Maybe the sound files are not linked correctly or the linking has changed?

That is a possibility, I don't think this event has ever worked in TrinityCore. Shouldn't be too hard to implement.

**Comment by Killyana** [comment]
It's already script https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/World/go_scripts.cpp#L1897

**Comment by illfated** [comment]
OK, thank you for linking the script, it reminded me of something slightly related: Timing in scripts has sometimes been broken in recent years, maybe due to core changes (like in the Ring of Law in Blackrock Depths). I haven't really thought about this Hourly Bells event at all during my years of using the TC environment, so I can't confirm or disprove whether it has ever worked or not.

My next idea is that maybe the script timing either does not match the game time or real time, so the sound files don't get played because of the time diff?
(I don't really know how the time calculation in this script works, so before anyone wants to tell me that I'm "talking out of my @$$", this is just meant as a thought pointing somewhere in the direction of finding out why it does not work.)

**Comment by jackpoz** [comment]
please try after https://github.com/TrinityCore/TrinityCore/commit/d9ae19d974a8026b0fd4eb8b1d11fa59c758f5d1

**Comment by Killyana** [comment]
All bells works correctly but not the one in Karazhan  https://woehead.way-of-elendil.fr/?object=182064

**Comment by jackpoz** [comment]
isn't there a known issue about game events not spawning objects in instances ?

**Comment by Kittnz** [comment]
not that i know of

**Comment by Wyrserth** [comment]
Yep, event spawns are not added to instances. They're skipped altogether right now because it's not trivial (they need to be added to any currently active instance as well as any newly created one).

Killyana mentioned it [here](https://github.com/TrinityCore/TrinityCore/commit/234674f1661906d994d1803150e812ce87c53b26), not sure if there's an issue open about it.

**Comment by Treeston** [comment]
gameevent spawn rewrite is on my list after #23603 gets merged.

gameevent spawns more-or-less just duplicate the spawn group system (and the quest pool system).

**Comment by Killyana** [comment]
I encountered the same issue with https://github.com/TrinityCore/TrinityCore/issues/18919#issuecomment-499943659

**Comment by illfated** [comment]
TrinityCore rev. 4074cb0f01d1 2019-09-01 19:05:29 +0200 (3.3.5 branch) (Unix, RelWithDebInfo, Static)
(https://tcubuntu.northeurope.cloudapp.azure.com)
Orgrimmar bells are drums, by the sound of it. Tonight, the "bell" told 10 drum beats at 10 pm (22:00 server time).

**Comment by Kittnz** [comment]
So i works 👍

**Comment by sirikfoll** [comment]
Is there any need for the Karazhan Bell to be linked to the game_event? If changed to a permanent spawn it will work too.

**Comment by Rushor** [comment]
do they need to be set to active?

**Comment by LordWaycrest** [comment]
Hi there,

I’m not sure, I’ve not been active with the project for a while now. As far as I know the event is working now.

Kind regards.

Lewis Brumby – Reptile Breeder and Home Hobbyist
Home – 01530 480405
Mobile – 07305 786336

Think before you print. Save energy and paper. Do you really need to print this email?

This e-mail and and files transmitted with it are confidential. If you are not the intended recipient, any reading, printing, storage, disclosure, copying or any other action taken in respect of this e-mail is prohibited and may be unlawful. If you are not the intended recipient, please notify me immediately by using the reply function and then permanently delete what you have received. The contents of this e-mail may have to be disclosed  to a request under the Data Protection Act 1998. Attachments to e-mail messages may contain viruses that may damage your system. Whilst I have taken every reasonable precauction to minimize this risk, I cannot accept any liabiltiy for any damage which you sustain as a result of these factors. You are advised to carry out your own virus checks before opening any attachment.

**Comment by Kittnz** [comment]
@Rushor yes, because the logic would pause otherwise and not finish all the bell sounds. If you could get close again say 20 min later, the bells will continue to go off when they shouldn't happen.

**Comment by shamage** [comment]
On version : TrinityCore rev. b0db728c49f0 2020-01-23 15:33:09 +0100 (3.3.5 branch) (Win64, RelWithDebInfo, Static) I tested it today ( tested in time 18:00 ==>> server time 18:00 )

**Comment by Aokromes** [comment]
updated title to make clear most of bug fixed.

**Comment by shamage** [comment]
I checked some information - 
I found the information about Karazhan Bell, but the location where it takes place is unknown . 
I found only this : [Karazhan Bell](https://wotlk.evowow.com/?event=73)
does anyone know more exactly where it should take place ?

**Comment by offl** [comment]
Duplicate of #26659

**Comment by offl** [comment]
Don't spawn that object every hour, spawn it always without linking to game event AS IT IS SUPPOSED TO BE and play sound if event is active.
